ashmem/ashmem.c

```diff
@@ -390,7 +390,7 @@ static int ashmem_mmap(struct file *file, struct vm_area_struct *vma)
 		ret = -EPERM;
 		goto out;
 	}
-	vma->vm_flags &= ~calc_vm_may_flags(~asma->prot_mask);
+	vm_flags_clear(vma, calc_vm_may_flags(~asma->prot_mask));
```

Comment:
Thank you very much for your patch :-) . And sorry for the late review.. :-(

vm_flags_clear() is unfortunately only available in kernel 6.3 and higher, so your commit will break the build for users that build against an older kernel version. I think you'd need conditional handling here, something like
```
#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0)
    vm_flags_clear(vma, calc_vm_may_flags(~asma->prot_mask));
#else
    vma->vm_flags &= ~calc_vm_may_flags(~asma->prot_mask);
#endif
```